qiskit/dagcircuit/_dagnode.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+   Object to represent the information at a node in the DAGCircuit
+"""
+
+
+class DAGNode:
+    """
+    Object to represent the information at a node in the DAGCircuit
+
+    It is used as the return value from *_nodes() functions and can
+    be supplied to functions that take a node.
+
+    """
+    def __init__(self, node_id=None, data_dict=None):
+        """ Create a node """
+        self.node_id = node_id
+
+        self.data_dict = data_dict
+
+    @property
+    def type(self):
+        """ Returns the type of the node else None"""
+        return self.data_dict['type'] if self.data_dict and'type' in self.data_dict else None
```

Comment:
If the comment about `__init__` sounds reasonable, you can take advantage of it and convert this line into:
```
return self.data_dict.get('type')
```
and similar variations for the rest of the properties.

Also, could you take the chance to document the type of the returned values in the docstring, for readability?

Reply:
That is much tidier, thanks!